### PR TITLE
Feature/redistreams

### PR DESCRIPTION
**Pull request by robert-dziewonski**

### CODE REVIEW

File: src/main/scala/com/softwaremill/mqperf/mq/RedisStreamsMq.scala [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+  override def createReceiver() = new MqReceiver {
+    private val client = jedisPool.getResource
+
+    override def receive(maxMsgCount: Int): List[(StreamEntryID, String)] = {
+      val readParams = new XReadGroupParams().count(maxMsgCount)
+      val stream = mapAsJavaMap(Map(StreamId -> UNRECEIVED_ENTRY))
+      Option(client.xreadGroup(ConsumerGroup, ConsumerId, readParams, stream))
+        .map(x => x.asScala.toList)
+        .getOrElse(List.empty)
```
Comment (adamw) [inline src/main/scala/com/softwaremill/mqperf/mq/RedisStreamsMq.scala:45]: [minor] these two lines can be also written as: `.flatMap(_.asScala.toList)`

File: src/main/scala/com/softwaremill/mqperf/mq/RedisStreamsMq.scala [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    }
+  }
+
+  override def createReceiver() = new MqReceiver {
+    private val client = jedisPool.getResource
+
+    override def receive(maxMsgCount: Int): List[(StreamEntryID, String)] = {
+      val readParams = new XReadGroupParams().count(maxMsgCount)
+      val stream = mapAsJavaMap(Map(StreamId -> UNRECEIVED_ENTRY))
+      Option(client.xreadGroup(ConsumerGroup, ConsumerId, readParams, stream))
```
Comment (adamw) [inline src/main/scala/com/softwaremill/mqperf/mq/RedisStreamsMq.scala:43]: `client.xreadGroup` can return `null`?

File: src/main/scala/com/softwaremill/mqperf/mq/RedisStreamsMq.scala [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+import redis.clients.jedis.params.XReadGroupParams
+import redis.clients.jedis.{HostAndPort, JedisPool, StreamEntryID}
+
+import java.util.UUID.randomUUID
+import scala.collection.JavaConverters._
+
+class RedisStreamsMq(testConfig: TestConfig) extends Mq with StrictLogging{
+  override type MsgId = StreamEntryID
+
+  final val StreamId = "mystream"
```
Comment (adamw) [inline src/main/scala/com/softwaremill/mqperf/mq/RedisStreamsMq.scala:15]: [minor] you usually don't need `final` here, unless you want to explicitly prohibit overriding. It's an immutable value either way

File: src/main/scala/com/softwaremill/mqperf/mq/RedisStreamsMq.scala (hunk 2 above)
Comment (robert-dziewonski) [reply src/main/scala/com/softwaremill/mqperf/mq/RedisStreamsMq.scala:43]: it does for the empty stream, pretty weird behaviour :)

Comment (adamw) [reply src/main/scala/com/softwaremill/mqperf/mq/RedisStreamsMq.scala:43]: ok, well APIs are sometimes surprising ;)

**Pull request merged**